components/ee/agent-smith/pkg/agent/agent.go

```diff
@@ -357,6 +400,48 @@ func (agent *Smith) Start(callback func(InfringingWorkspace, []PenaltyKind)) {
 	}
 }
 
+func (agent *Smith) cleanupDeadPIDS(ctx context.Context) {
+	ticker := time.NewTicker(30 * time.Second)
```

Comment:
nit: missing `defer ticket.Stop()`

Reply:
This is in the ctx.Done, however since the goroutine is being moved out this needs to exist now.